llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -980,7 +980,28 @@ getDeviceCodeSplitter(ModuleDesc &&MD, IRSplitMode Mode, bool IROutputOnly,
     // output files in existing tests.
     Categorizer.registerSimpleStringAttributeRule("sycl-register-alloc-mode");
     Categorizer.registerSimpleStringAttributeRule("sycl-grf-size");
-    Categorizer.registerListOfIntegersInMetadataSortedRule("sycl_used_aspects");
+    Categorizer.registerRule([&](Function *F) {
+      std::string Result = "";
```

Comment:
I think that [`SmallString`](https://llvm.org/doxygen/classllvm_1_1SmallString.html) would be a better fit here to reduce amount of re-allocations on `operator+=`